app/javascript/packages/document-capture/components/document-capture.jsx

```diff
@@ -17,12 +19,23 @@ import useI18n from '../hooks/use-i18n';
  *                                    Defaults to false.
  */
 
+/**
+ * Returns error messages interspersed with line break React element.
+ *
+ * @param {string[]} errors Error messages.
+ *
+ * @return {ReactNode[]} Formatted error messages.
+ */
+export function getFormattedErrors(errors) {
+  return errors.flatMap((error, i) => [<br key={i} />, error]).slice(1);
```

Comment:
is the `key` just to be able to tell these apart?

Reply:
> is the `key` just to be able to tell these apart?

It's to appease React 😄  

```diff
diff --git a/app/javascript/packages/document-capture/components/document-capture.jsx b/app/javascript/packages/document-capture/components/document-capture.jsx
index 9afe466e7..104b5efbc 100644
--- a/app/javascript/packages/document-capture/components/document-capture.jsx
+++ b/app/javascript/packages/document-capture/components/document-capture.jsx
@@ -30 +30 @@ export function getFormattedErrors(errors) {
-  return errors.flatMap((error, i) => [<br key={i} />, error]).slice(1);
+  return errors.flatMap((error) => [<br />, error]).slice(1);
```

```
$ yarn mocha spec/javascripts/packages/document-capture/components/document-capture-spec.jsx
yarn run v1.22.4
[...]

  1) "after each" hook for "renders handled submission failure":
     AssertionError: Unexpected console logging: Warning: Each child in a list should have a unique "key" prop.%s%s See https://fb.me/react-warning-keys for more information.%s: expected [ Array(1) ] to be undefined
[...]
```

Related: https://reactjs.org/docs/lists-and-keys.html#keys

Index is a pretty lazy approach (as mentioned in the documentation), but it shouldn't be too important in this situation. Strings aren't an issue either, hence why we don't need to do anything about `error`. I don't know exactly how it works internally, but I assume the string can sorta act as its own key. This was partly why I chose index over `error` as `key`, since both could technically have worked, but I wasn't sure if there might be a chance for conflict if internally `[<br key={error} />, error]` would be the same key 🤷..

There's also an ESLint rule for this in the React ESLint plugin we're using. I'm actually a bit surprised it's not enabled by default. Might make sense to enable it!

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-key.md